### DISCUSSION TITLE
Comparing next_rate in Agile with an input_number in automation

### DISCUSSION THREAD

**Discussion by willenic** [BottlecapDave/HomeAssistant-OctopusEnergy#421] [Q&A] [answered]
Hi All,

I'm trying to use this integration in Home Assistant by comparing the state of the octopus next_rate sensor. 

I want to turn my pod point charger on and off when my Octopus Agile tariff goes below or above a price I input. It's easier for me to set a number by looking at the day ahead and know how well charged the car is than choose one of Dave's intermittent/constant Target rate sensors. How charged my car is and my thoughts on future price affect this.

So, I created an automation looking for a **trigger** of a change in the next_rate below 1p/kWh (£0.01 per kWh), with a **condition** to check that the _next_rate_ is below an _input_number_ which is a user defined variable set by a helper called _"rate_to_charge"._  If the trigger and condition are both true I call a service after a delay to set my pod point charger charging.  I was going to set another automation with reverse logic to turn it off.

My condition was firing as true when I check in the template tool but the Trigger obviously isn't triggering. Earlier today the next rate was shown as 0 or negative so I'd have expected the trigger to fire.  Here's my YAML

```alias: Charge in 28 mins if next rate < input rate
description: ""
trigger:
  - type: monetary
    platform: device
    device_id: my_device
    entity_id: my_entity
    domain: sensor
    below: 0.01
condition:
  - condition: template
    value_template: >-
      {% if
      states('sensor.octopus_energy_electricity_accountnum_mpannum_next_rate')
      < states('input_number.rate_to_charge') %}
        True
      {% else %}
        False
      {% endif %}
action:
  - delay:
      hours: 0
      minutes: 28
      seconds: 0
      milliseconds: 0
  - service: pod_point.charge_now
    data:
      hours: 4
      minutes: 0
      seconds: 0
      config_entry_id: config_id
mode: single```

**Comment by willenic** [comment]
And here's the Visual editor for the trigger.  Any clues why the trigger isn't kicking in?

![Screenshot 2023-09-24 193950](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/144794340/10fb004c-cb77-4ae7-81ee-7527c9255089)

**Reply by BottlecapDave** [reply]
Hello. I assume you're not getting any traces against the automation.

I personally usually use [numeric state](https://www.home-assistant.io/docs/automation/trigger/#numeric-state-trigger) triggers, however you should note that if the sensor is below 0.01 and then is updated to another value below 0.01, then the automation won't re-trigger. This might apply to the trigger you're currently using, and might be why it didn't trigger for you before?

If that doesn't work, then you could do a standard [state trigger](https://www.home-assistant.io/docs/automation/trigger/#state-trigger) with no restrictions, which will trigger whenever the sensor changes, and move the below 0.01 checks to be conditions.

**Reply by willenic** [reply]
Thanks, I'll have a play.  The trace result from last night is as below but that's as I expected at that time. I'll check if there is a history on the others, the result was showing false when the condition. IIRC is was getting a false result on the trace when the template result was showing true.

```Stopped because a condition failed at September 24, 2023 at 9:00:40 PM (runtime: 0.00 seconds```

**Reply by BottlecapDave** [reply, marked as answer]
For you condition, you should be able to just do

```yaml
condition:
  - condition: template
    value_template: >-
      {{ states('sensor.octopus_energy_electricity_accountnum_mpannum_next_rate')
      < states('input_number.rate_to_charge') }}
```

just in case it doesn't like your boolean results for some reason

**Comment by BottlecapDave** [comment]
Also if you're looking for something adventurous with the target rate sensors, you can see how people are using the update target rate config service to dynamically adjust the target rates for their batteries - https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/112
